app/controllers/graders_controller.rb

```diff
@@ -87,6 +87,44 @@ def global_actions
       end
     end
 
+    # '%w' syntax is a Ruby idiom that, preferred over ["...", "...", ...] syntax for its conciseness and readability
+    # It represents an array of strings
+    if %w[assign random_assign].include? params[:global_actions]
+      inactive_graders = [] # used to filter the 'grader_ids' array and to throw a custom flash message error
+
+      # Solving N+1 querying problem by executing the SQL query:
+      # SELECT * FROM roles WHERE id IN (..., ..., ..., ...)
+      # (thus making one database request in the beginning instead of one for each grader id)
+      graders = Ta.where(id: grader_ids)
```

Comment:
This query can be improved in a few ways:

- use an association to only get the TAs associated with the `current_course`
- add an additional argument to the `where` to only get inactive graders
- use `pluck` to get the resulting `id`s and `user_name`s in an array. This is the value that you want for `inactive_grader_ids`. This saves instantiating `Ta` objects unnecessarily.